src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -76,32 +81,139 @@ public void run(Connection conn) throws SQLException {
+                break;
+              case BINOMIAL:
+                generatedFloat = param.getNextFloatBinomial();
+                break;
+              default:
+                throw param.createRuntimeException(paramType);
+            }
+            stmt.setFloat(i + 1, generatedFloat);
+            break;
+          case "bigint":
+            Long generatedLong;
+            switch (distribution) {
+              case UNIFORM:
+                generatedLong = param.getNextLongUniform();
+                break;
+              case BINOMIAL:
+                generatedLong = param.getNextLongBinomial();
+                break;
+              case ZIPFIAN:
+                generatedLong = param.getNextLongZipf();
+                break;
+              case SCRAMBLED:
+                generatedLong = param.getNextLongScrambled();
+                break;
+              default:
+                throw param.createRuntimeException(paramType);
+            }
+            stmt.setLong(i + 1, generatedLong);
+            break;
+          case "varchar":
+          case "string":
+            switch (distribution) {
+              case UNIFORM:
+                stmt.setString(i + 1, param.getNextString());
+                break;
+              default:
+                throw param.createRuntimeException(paramType);
+            }
+            break;
+          case "timestamp":
+          case "date":
+          case "time":
+            Long generatedTimestamp;
+            switch (distribution) {
+              case UNIFORM:
+                generatedTimestamp = param.getNextLongUniform();
+                break;
+              case BINOMIAL:
+                generatedTimestamp = param.getNextLongBinomial();
+                break;
+              case ZIPFIAN:
+                generatedTimestamp = param.getNextLongZipf();
+                break;
+              case SCRAMBLED:
+                generatedTimestamp = param.getNextLongScrambled();
+                break;
+              default:
+                throw param.createRuntimeException(paramType);
+            }
+            if (paramType.equals("timestamp")) {
+              stmt.setTimestamp(i + 1, new Timestamp(generatedTimestamp));
+            } else if (paramType.equals("date")) {
+              stmt.setDate(i + 1, new Date(generatedTimestamp));
+            } else {
+              stmt.setTime(i + 1, new Time(generatedTimestamp));
+            }
+            break;
+          default:
+            throw new RuntimeException(
+                "Support for distributions for the type: "
+                    + paramType
+                    + " is current not implemented");
+        }
+
       } else {
         try {
-          // TODO: add support for nullable other types
-          // For instance, can we provide a <value /> tag in the XML file to represent a
-          // NULL value?
-          // Or does it need a special marker like "$null" to signify a NULL value?
```

Comment:
Do we have tests for NULL value handling now?

Reply:
Yes, the system test how to handle multiple cases (Explicitly stating a value to be of type NULL or having an empty <value /> tag).